lib/endpoints/boot_params.js

```diff
@@ -47,16 +48,24 @@ function validateBootModules(boot_modules) {
             continue;
         }
 
+        if (/[.]+/.test(path.dirname(bm.path))) {
```

Comment:
That also matches a component that looks like `...` or `some.dir`.  It also seems that the last component being a `.` or `..` should be invalid. Perhaps better as:

```
paths = path.split('/');
if (paths.indexOf('.') != -1 || paths.indexOf('..') != -1)
```

Reply:
I think I've got a better and simpler approach. Use `path.normalize` and compare against provided path. If those aren't exactly the same, complain. If for some reason that's not working correctly, fill a bug for nodejs ;-)

Reply:
Oh, also note I've moved the whole validation function to `lib/validation/endpoints.js` which seems to be the correct place.